string url
string course
---
bool was_set
